apps/browser/qml/pages/components/AddHomePageDialog.qml

```diff
@@ -0,0 +1,63 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+import Sailfish.Policy 1.0
+import org.nemomobile.configuration 1.0
+
+Dialog {
+    id: root
+    property alias text: textField.text
+    property alias placeholderText: textField.placeholderText
+    property alias description: textField.description
+
+    function containsSpaces(text) {
+       return /^\s+|\s+$|\s*(\s)/.test(text) && text.length > 0
+    }
+
+    canAccept: (textField.text !== "" && !textField.errorHighlight)
```

Comment:
As far as I can tell the `textField.text !== "" ` condition is already covered by `containsSpaces()` so can be removed from here.

Reply:
containsSpaces will be relevant when `text.length > 0`

Reply:
Yes, you're absolutely right, my mistake. Thanks for clarifying.

In this case, maybe the error highlight should show if the textField is empty, in order to clarify to the user why the dialogue can't be accepted. What do you think?

Reply:
@llewelld  `In this case, maybe the error highlight should show if the textField is empty, in order to clarify to the user why the dialogue can't be accepted. What do you think?` I added to the description: ` //% "Enter website address that should be shown when browser is opened with no tabs to load."
        description: !errorHighlight ? qsTrId("sailfish_browser-he-home_page_website-address")
                                       //% "Spaces can't be included in the URL."
                                     : qsTrId("sailfish_browser-he-home_page_website-spaces_error")`